### PR TITLE
Separated Integration Test Files

### LINKED ISSUE #203: Split integration tests into separate files
`integration.test.ts` is getting large and unwieldy. The tests should be separated into different files, based on section (e.g. `download_upload.integration.test.ts`, `file.integration.test.ts`, etc.). They should be in their own directory e.g. `integration`. The tests should continue to run on the desired portal set in `SKYNET_JS_INTEGRATION_TEST_SERVER`, if provided.

### PR DESCRIPTION
**Pull request by AmberMichaelis**
# PULL REQUEST

## Issues Closed

Closes #203

### CODE REVIEW

**Review by m-cat — APPROVED** [review]
Fantastic work!

**Review by m-cat — APPROVED** [review]
Fabulous work!

File: src/utils/url.test.ts [hunk 1]
```diff
@@ -7,14 +7,14 @@ const skylink = "XABvi7JtJbQSMAcDwnUnmp2FKDPjg8_tTTFP4BwMSxVdEg";
 const skylinkBase32 = "bg06v2tidkir84hg0s1s4t97jaeoaa1jse1svrad657u070c9calq4g";
 
 describe("addUrlQuery", () => {
-  const parts: Array<[string, Record<string, unknown>, string]> = [
+  const parts: Array<[string, { [key: string]: string | undefined }, string]> = [
```
Comment (kwypchlo) [inline src/utils/url.test.ts:10]:
```suggestion
  const parts: Array<[string, { [key: string]: string | number | boolean }, string]> = [
```

we should keep supporting all basic types - namely numbers and booleans

File: src/utils/url.ts [hunk 2]
```diff
@@ -85,7 +85,7 @@ export function addSubdomain(url: string, subdomain: string): string {
  * @param query - The query parameters.
  * @returns - The final URL.
  */
-export function addUrlQuery(url: string, query: Record<string, unknown>): string {
+export function addUrlQuery(url: string, query: { [key: string]: string | undefined }): string {
```
Comment (kwypchlo) [inline src/utils/url.ts:88]:
```suggestion
export function addUrlQuery(url: string, query: { [key: string]: string | number | boolean }): string {
```

is undefined possible here? in any case we should keep supporting all basic types - namely numbers and booleans included

File: src/upload.ts [hunk 3]
```diff
@@ -354,7 +354,7 @@ export async function uploadDirectoryRequest(
     formData.append(PORTAL_DIRECTORY_FILE_FIELD_NAME, file as File, path);
   });
 
-  const query: Record<string, unknown> = { filename };
+  const query: { [key: string]: string | undefined } = { filename };
```
Comment (kwypchlo) [inline src/upload.ts:357]:
```suggestion
  const query: { [key: string]: string | number | boolean } = { filename };
```

File: src/registry.ts [hunk 4]
```diff
@@ -297,7 +297,7 @@ export function getEntryUrlForPortal(
   const query = {
     publickey: ensurePrefix(publicKey, ED25519_PREFIX),
     datakey: dataKeyHashHex,
-    timeout: DEFAULT_GET_ENTRY_TIMEOUT,
+    timeout: DEFAULT_GET_ENTRY_TIMEOUT.toString(),
```
Comment (kwypchlo) [inline src/registry.ts:300]:
```suggestion
    timeout: DEFAULT_GET_ENTRY_TIMEOUT,
```

`makeUrl` should be happy to accept numbers

File: src/download.ts [hunk 5]
```diff
@@ -293,9 +293,9 @@ export async function getHnsUrl(
 
   const opts = { ...DEFAULT_DOWNLOAD_HNS_OPTIONS, ...this.customOptions, ...customOptions };
 
-  const query: Record<string, unknown> = {};
+  const query: { [key: string]: string | undefined } = {};
```
Comment (kwypchlo) [inline src/download.ts:296]:
```suggestion
  const query: { [key: string]: string | number | boolean } = {};
```

File: src/download.ts [hunk 6]
```diff
@@ -293,9 +293,9 @@ export async function getHnsUrl(
...
   if (opts.download) {
-    query.attachment = true;
+    query.attachment = "true";
```
Comment (kwypchlo) [inline src/download.ts:298]:
```suggestion
    query.attachment = true;
```

`addUrlQuery` should be happy to accept numbers and booleans

**Review by kwypchlo — CHANGES_REQUESTED** [review]
I started doing a code review from the bottom for some reason and started to point out all `{ [key: string]: string | undefined }` types - we should not enforce the value to be a string, we should keep accepting number and boolean, otherwise it's a breaking change, unnecessary too

File: src/download.ts [hunk 7]
```diff
@@ -220,10 +220,10 @@ export function getSkylinkUrlForPortal(
 
   const opts = { ...DEFAULT_DOWNLOAD_OPTIONS, ...customOptions };
 
-  const query: Record<string, unknown> = {};
+  const query: { [key: string]: string | undefined } = {};
   if (opts.download) {
     // Set the "attachment" parameter.
-    query.attachment = true;
+    query.attachment = "true";
```
Comment (kwypchlo) [inline src/download.ts:226]:
```suggestion
    query.attachment = true;
```

File: src/download.ts [hunk 8]
```diff
@@ -220,10 +220,10 @@ export function getSkylinkUrlForPortal(
 
   const opts = { ...DEFAULT_DOWNLOAD_OPTIONS, ...customOptions };
 
-  const query: Record<string, unknown> = {};
+  const query: { [key: string]: string | undefined } = {};
```
Comment (kwypchlo) [inline src/download.ts:223]:
```suggestion
  const query: { [key: string]: string | number | boolean } = {};
```

File: src/client.ts [hunk 9]
```diff
@@ -308,7 +309,7 @@ export async function buildRequestUrl(
   endpointPath: string,
   url?: string,
   extraPath?: string,
-  query?: Record<string, unknown>
+  query?: { [key: string]: string | undefined }
```
Comment (kwypchlo) [inline src/client.ts:312]:
```suggestion
  query?: { [key: string]: string | number | boolean }
```

File: src/client.ts [hunk 10]
```diff
@@ -69,7 +69,7 @@ export type RequestConfig = CustomClientOptions & {
   url?: string;
   method?: Method;
   headers?: Headers;
-  query?: Record<string, unknown>;
+  query?: { [key: string]: string | undefined };
```
Comment (kwypchlo) [inline src/client.ts:72]:
```suggestion
  query?: { [key: string]: string | number | boolean };
```

**Review by kwypchlo — CHANGES_REQUESTED** [review]

**Comment by m-cat** [comment]
For reference:

I made this change for them because one of the dependencies (`url-parse` I think) got updated (or the types got updated) and they were getting a lint error. I figure that fixing all the internal types would be fine because it's only internal. Otherwise I would have had to do a `// @ts-ignore` and I'm trying to minimize those, as much as possible.

**Review by kwypchlo — DISMISSED** [review]
ok I think overall it's fine, I had to read the source code thoroughly but it seems like in fact those methods that accept string values are not exposed anywhere, still feels weird and unnatural but works

**Pull request merged**